src/hammer-vlsi/hammer_vlsi/hammer_build_systems.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+#  hammer_build_systems.py
+#  Class containing all the methods to create VLSI flow build system infrastructure
+#
+#  See LICENSE for licence details.
+
+from .driver import HammerDriver
+
+import os
+import sys
+import textwrap
+from typing import List, Dict, Tuple, Callable
+
+def build_noop(driver: HammerDriver) -> dict:
+    """
+    Do nothing, just return the dependency graph.
+
+    :param driver: The HammerDriver
+    :return: The dependency graph
+    """
+    dependency_graph = driver.get_hierarchical_dependency_graph()
+    return dependency_graph
+
+
+def build_makefile(driver: HammerDriver) -> dict:
+    """
+    Build a Makefile include in the obj_dir called hammer.d. This is intended to be dynamically
+    created and included into a top-level Makefile.
+
```

Comment:
Could you describe a bit about the Makefile that is generated at a high level? Looking at the unit tests I see for example `redo-` targets but couldn't find a high level description of what they are or what they do (don't see it in dependency graph either)

Reply:
Sure, I agree it needs some more meat here.

Reply:
Done